cpp/include/legate_dataframe/core/column.hpp

```diff
@@ -200,17 +250,33 @@ class LogicalColumn {
   /**
    * @brief Returns the number of rows
    *
-   * @throw std::runtime_error if column is unbound.
+   * @throw std::runtime_error if column is unbound. std::logic_error if a scalar
+   * but the number of rows isn't one.
    * @return The number of rows
    */
   [[nodiscard]] size_t num_rows() const
   {
     if (unbound()) {
       throw std::runtime_error("Cannot call `.num_rows()` on a unbound LogicalColumn");
     }
-    return array_->volume();
+    auto rows = array_->volume();
+    if (scalar() && rows != 1) {
+      throw std::logic_error("PhysicalColumn is scalar but doesn't have one row.");
+    }
+    return rows;
   }
 
+  /**
+   * @brief Return true if the column is considered scalar.
+   *
+   * A scalar column always has a size of one.  Scalar columns are mostly supported
+   * for binary operations and have scalar/broadcast semantics if supported.
+   * (For unsupported operations they may work, but behave like a length 1 column).
+   *
+   * @return true if scalar, otherwise false.
+   */
+  bool scalar() const { return scalar_; };
```

Comment:
nit: maybe rename to `is_scalar`, to make it clear that it doesn't return a _scalar_ ?

Reply:
Thanks for the nudge, done.  Had wondered briefly but not acted up on it :).